### PR TITLE
ci: using uv to setup python and install dependencies for github actions

### PR DESCRIPTION
**Pull request by timurcarstensen**
**changes**
- uses `uv` for `venv` and installing dependencies in our github actions workflows
- switches to [PEP 621](https://peps.python.org/pep-0621/) for `pyproject.toml` and discards of `poetry`

**todo**
- [x] update docs 
- [x] update other workflows
- [x] Add a `uv build` line to the contributing docs for the publishing part

### CODE REVIEW

**Review by eddiebergman — COMMENTED** [review]
Seems good at a glance although you've have to update the documentation/contributing guide as well. I recommend searching up for the string "poetry" to find sources of it.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -344,7 +320,7 @@ To publish to PyPI:
 3. Run
 
 ```bash
-poetry publish --build
+poetry publish
```
Comment (eddiebergman) [inline CONTRIBUTING.md]: Still running `poetry` for publishing?

I've been used to running `twine upload dist/*` as a standalone way. Perhaps `uv` has some option for this?

**Review by eddiebergman — COMMENTED** [review]
Generally big fan, just want to check that `pyproject.toml` gets updated to remove peotry?

Also would mean `poetry publish` would cease to work

File: CONTRIBUTING.md (hunk 1 above)
Comment (timurcarstensen) [reply CONTRIBUTING.md]: Ah this is outdated, I switched to `uv publish` but haven't had the chance to test this yet

Comment (timurcarstensen) [reply CONTRIBUTING.md]: Can we test this without directly creating a new release 😅

**Review by eddiebergman — APPROVED** [review]

**Pull request merged**